include/swift/Basic/ImmutablePointerSet.h

```diff
@@ -112,7 +112,8 @@ template <typename T> class ImmutablePointerSet : public llvm::FoldingSetNode {
     return *LowerBound == Ptr;
   }
 
-  using iterator = typename decltype(Data)::iterator;
+  // Ideally we would use decltype(Data)::iterator, but MSVC reports an error
```

Comment:
Nitpick: comments should end with periods. (Although I don't really think you need this one; the alternate formulation isn't so bad that anyone would be tempted to change it.)